<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Python: module math_exception</title>
</head><body>

<table class="heading">
<tr class="heading-text decor">
<td class="title">&nbsp;<br><strong class="title">math_exception</strong></td>
<td class="extra"><a href=".">index</a><br><a href="file:c%3A%5Cusers%5Cshelv%5Cdesktop%5C%D0%BD%D0%B0%D0%B2%D1%87%D0%B0%D0%BD%D0%BD%D1%8F%20%28%D0%BE%D0%BC%D0%B3%29%5Cspeclang%5Cclasses%5Cexceptions%5Cmath_exception.py">c:\users\shelv\desktop\навчання (омг)\speclang\classes\exceptions\math_exception.py</a></td></tr></table>
    <p></p>
<p>
<table class="section">
<tr class="decor index-decor heading-text">
<td class="section-title" colspan=3>&nbsp;<br><strong class="bigsection">Classes</strong></td></tr>
    
<tr><td class="decor index-decor"><span class="code">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></td><td>&nbsp;</td>
<td class="singlecolumn"><dl>
<dt class="heading-text"><a href="builtins.html#object">builtins.object</a>
</dt><dd>
<dl>
<dt class="heading-text"><a href="math_exception.html#CalculatorError">CalculatorError</a>
</dt></dl>
</dd>
</dl>
 <p>
<table class="section">
<tr class="decor title-decor heading-text">
<td class="section-title" colspan=3>&nbsp;<br><a name="CalculatorError">class <strong>CalculatorError</strong></a>(<a href="builtins.html#object">builtins.object</a>)</td></tr>
    
<tr><td class="decor title-decor" rowspan=2><span class="code">&nbsp;&nbsp;&nbsp;</span></td>
<td class="decor title-decor" colspan=2><span class="code"><a href="#CalculatorError">CalculatorError</a>(error_type,&nbsp;message)<br>
&nbsp;<br>
Custom&nbsp;exception&nbsp;class&nbsp;for&nbsp;calculator&nbsp;errors.<br>
&nbsp;<br>
Attributes:<br>
&nbsp;&nbsp;&nbsp;&nbsp;error_type&nbsp;(type):&nbsp;The&nbsp;type&nbsp;of&nbsp;the&nbsp;exception.<br>
&nbsp;&nbsp;&nbsp;&nbsp;message&nbsp;(str):&nbsp;The&nbsp;error&nbsp;message.<br>
&nbsp;<br>
Methods:<br>
&nbsp;&nbsp;&nbsp;&nbsp;__str__:&nbsp;Returns&nbsp;a&nbsp;string&nbsp;representation&nbsp;of&nbsp;the&nbsp;error.<br>&nbsp;</span></td></tr>
<tr><td>&nbsp;</td>
<td class="singlecolumn">Methods defined here:<br>
<dl><dt><a name="CalculatorError-__init__"><strong>__init__</strong></a>(self, error_type, message)</dt><dd><span class="code">Initializes&nbsp;a&nbsp;<a href="#CalculatorError">CalculatorError</a>&nbsp;instance&nbsp;with&nbsp;the&nbsp;given&nbsp;error&nbsp;type&nbsp;and&nbsp;message.<br>
&nbsp;<br>
Args:<br>
&nbsp;&nbsp;&nbsp;&nbsp;error_type&nbsp;(type):&nbsp;The&nbsp;type&nbsp;of&nbsp;the&nbsp;exception.<br>
&nbsp;&nbsp;&nbsp;&nbsp;message&nbsp;(str):&nbsp;The&nbsp;descriptive&nbsp;message&nbsp;for&nbsp;the&nbsp;exception.</span></dd></dl>

<dl><dt><a name="CalculatorError-__str__"><strong>__str__</strong></a>(self)</dt><dd><span class="code">Returns&nbsp;the&nbsp;string&nbsp;representation&nbsp;of&nbsp;the&nbsp;<a href="#CalculatorError">CalculatorError</a>.<br>
&nbsp;<br>
Combines&nbsp;the&nbsp;error&nbsp;type&nbsp;and&nbsp;message&nbsp;into&nbsp;a&nbsp;formatted&nbsp;string.<br>
&nbsp;<br>
Returns:<br>
&nbsp;&nbsp;&nbsp;&nbsp;str:&nbsp;A&nbsp;formatted&nbsp;error&nbsp;message.</span></dd></dl>

<hr>
Data descriptors defined here:<br>
<dl><dt><strong>__dict__</strong></dt>
<dd><span class="code">dictionary&nbsp;for&nbsp;instance&nbsp;variables&nbsp;(if&nbsp;defined)</span></dd>
</dl>
<dl><dt><strong>__weakref__</strong></dt>
<dd><span class="code">list&nbsp;of&nbsp;weak&nbsp;references&nbsp;to&nbsp;the&nbsp;object&nbsp;(if&nbsp;defined)</span></dd>
</dl>
</td></tr></table></td></tr></table><p>
<table class="section">
<tr class="decor functions-decor heading-text">
<td class="section-title" colspan=3>&nbsp;<br><strong class="bigsection">Functions</strong></td></tr>
    
<tr><td class="decor functions-decor"><span class="code">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></td><td>&nbsp;</td>
<td class="singlecolumn"><dl><dt><a name="-exception"><strong>exception</strong></a>(exception)</dt><dd><span class="code">Helper&nbsp;function&nbsp;to&nbsp;create&nbsp;a&nbsp;<a href="#CalculatorError">CalculatorError</a>&nbsp;instance&nbsp;from&nbsp;an&nbsp;exception.<br>
&nbsp;<br>
Args:<br>
&nbsp;&nbsp;&nbsp;&nbsp;exception&nbsp;(Exception):&nbsp;The&nbsp;exception&nbsp;to&nbsp;convert&nbsp;into&nbsp;a&nbsp;<a href="#CalculatorError">CalculatorError</a>.<br>
&nbsp;<br>
Returns:<br>
&nbsp;&nbsp;&nbsp;&nbsp;<a href="#CalculatorError">CalculatorError</a>:&nbsp;A&nbsp;<a href="#CalculatorError">CalculatorError</a>&nbsp;instance&nbsp;encapsulating&nbsp;the&nbsp;given&nbsp;exception.</span></dd></dl>
</td></tr></table>
</body></html>